### PR TITLE
Update default version to latest LTS + add pam authentication

### PR DESCRIPTION
**Pull request by PierreR**

### CODE REVIEW

**Comment by racodond** [comment]
@PierreR Could you please also update the test classes to check on the latest LTS?

**Comment by PierreR** [comment]
My original intend was to make two separate PR. I hope it is alright as it is.

File: manifests/init.pp [hunk 1]
```diff
@@ -27,6 +27,8 @@
   $arch             = $sonarqube::params::arch,
   $https            = {},
   $ldap             = {},
+  # ldap and pam are mutually exclusive. Setting $ldap will annihilate the setting of $pam
```
Comment (racodond) [inline manifests/init.pp]: @PierreR I'm not sure to understand this comment. I don't see in the code how ldap can annihilate the setting of pam. Could you please clarify?

**Comment by racodond** [comment]
@PierreR Fine by me if everything is in the same PR.

File: manifests/init.pp (hunk 1 above)
Comment (PierreR) [inline manifests/init.pp]: If you look in the template file, I use an `elsif`. If you have some `ldap` setting you will never get the `pam` one. I believe this is more secure. The comment was meant to clarify this behavior.

Comment (racodond) [inline manifests/init.pp]: Got it now. Thanks.

**Comment by racodond** [comment]
Thanks for your contribution.

**Pull request merged**